project/shacl/modos_schema.shacl.ttl

```diff
@@ -126,291 +140,335 @@ modos:VariantSet a sh:NodeShape ;
-            sh:path modos:data_path ],
-        [ sh:datatype xsd:string ;
-            sh:description "A human-readable name for a thing" ;
+            sh:description "A string specifying the release or version of a software or resource." ;
             sh:maxCount 1 ;
-            sh:order 6 ;
-            sh:path schema1:name ],
-        [ sh:description "A unique identifier for a thing" ;
+            sh:nodeKind sh:Literal ;
+            sh:order 4 ;
+            sh:path modos:version ],
+        [ sh:datatype xsd:anyURI ;
+            sh:description "A unique identifier for a thing" ;
             sh:maxCount 1 ;
             sh:minCount 1 ;
+            sh:nodeKind sh:Literal ;
             sh:order 5 ;
             sh:path schema1:identifier ],
         [ sh:datatype xsd:string ;
-            sh:description "A string specifying the release or version of a software or resource." ;
+            sh:description "A human-readable description for a thing" ;
             sh:maxCount 1 ;
-            sh:order 4 ;
-            sh:path modos:version ] ;
+            sh:nodeKind sh:Literal ;
+            sh:order 7 ;
+            sh:path schema1:description ] ;
     sh:targetClass modos:ReferenceGenome .
 
 modos:Sample a sh:NodeShape ;
     sh:closed true ;
     sh:description "A biological sample used in assays. Examples include a whole organism, tissue or cell line." ;
     sh:ignoredProperties ( rdf:type ) ;
-    sh:property [ sh:datatype xsd:string ;
-            sh:description "The organization responsible for collecting a given sample." ;
-            sh:order 4 ;
-            sh:path modos:collector ],
-        [ sh:datatype xsd:integer ;
+    sh:property [ sh:datatype xsd:integer ;
             sh:description "The taxid number describing the taxonomic range of a sample." ;
+            sh:nodeKind sh:Literal ;
             sh:order 3 ;
             sh:path modos:taxon_id ],
         [ sh:datatype xsd:string ;
-            sh:description "The cell type name." ;
+            sh:description "The organization responsible for collecting a given sample." ;
+            sh:nodeKind sh:Literal ;
+            sh:order 4 ;
+            sh:path modos:collector ],
+        [ sh:datatype xsd:string ;
+            sh:description "A human-readable description for a thing" ;
             sh:maxCount 1 ;
-            sh:order 0 ;
-            sh:path modos:cell_type ],
+            sh:nodeKind sh:Literal ;
+            sh:order 7 ;
+            sh:path schema1:description ],
         [ sh:datatype xsd:string ;
             sh:description "A human-readable name for a thing" ;
             sh:maxCount 1 ;
+            sh:nodeKind sh:Literal ;
             sh:order 6 ;
             sh:path schema1:name ],
-        [ sh:description "A unique identifier for a thing" ;
+        [ sh:description "The cell type name or code, if applicable." ;
+            sh:maxCount 1 ;
+            sh:or ( [ sh:in [ rdf:rest () ] ] [ sh:datatype xsd:string ;
```

Comment:
not 100% sure what you're trying to say here, but 

`sh:in [rdf:rest () ] ` is basically saying the value for this property has to be an empty list, which is not really a thing. You could not make an instances which would be valid according to this part of the shape. Of course, since it's within the sh:or, it might not be a problem if the user gives a string, but if you want them to give uberon:somecode123 then it will trigger and fail. Probably the code generation fucked up slightly, as I would expect an rdf:list here as the range for sh:in as you did correctly for other properties. e.g. line 180.

`[ sh:in ( "Value1" "Value2" "Value3" ) ]`